pilot/pkg/model/service.go

```diff
@@ -625,6 +625,19 @@ func GetServiceAccounts(svc *Service, ports []int, discovery ServiceDiscovery) [
 	return sa.UnsortedList()
 }
 
+// GetServiceAccountsFromInstances returns service accounts for a service derived from passed in instances.
+func GetServiceAccountsFromInstances(svc *Service, instances []*ServiceInstance) []string {
```

Comment:
This is missing all the logic in the aggregate controller GetIstioServiceAccounts unless I am missing something. And if we don't need GetIstioServiceAccounts we should remove it.

Reply:
Ah.. That is a great catch. I just looked at kubecontroller and changed - AggregateController has some additional logic. Will fix it.